src/main/java/org/apache/sysml/hops/estim/EstimatorMatrixHistogram.java

```diff
@@ -71,15 +71,66 @@ public double estim(MMNode root) {
 		
 		return ret;
 	}
-
-	@Override
+	
+	@Override 
 	public double estim(MatrixBlock m1, MatrixBlock m2) {
 		MatrixHistogram h1 = new MatrixHistogram(m1, _useExcepts);
 		MatrixHistogram h2 = (m1 == m2) ? //self product
 			h1 : new MatrixHistogram(m2, _useExcepts);
 		return estimIntern(h1, h2);
 	}
 
+	public double estim(MatrixBlock m1, MatrixBlock m2, String op) {
```

Comment:
1) Please add an enum for the operation type to avoid this string handling.
2) Extend the estimator class to have overloaded estim method with and without op code
3) The default estim without opcode should simply call the other estim method with default opcode MM (matrix mult)